src/object/object_primitive.c

```diff
@@ -11212,11 +11304,20 @@ mr_readval_string_internal (OR_BUF * buf, DB_VALUE * value, TP_DOMAIN * domain,
 		  start = buf->ptr;
 		}		/* size != -1 */
 
-	      str_length = or_get_varchar_length (buf, &rc);
+	      /* Get the length of the string, be it compressed or uncompressed. */
+	      rc = or_get_varchar_compression_lengths (buf, &compressed_size, &decompressed_size);
 	      if (rc != NO_ERROR)
 		{
 		  return ER_FAILED;
 		}
+	      if (compressed_size <= 0)
```

Comment:
Since the `or_get_varchar_compression_lengths` sets `-1` to `compressed_size` if there was no compression made, `compressed_size` should never be `0`, because this would mean that there was a compression made with a length of `0` and that is quite not possible. You want me to add an `assert` to catch whether there is actually this case?